src/domains/profile/pages/CreateProfile/CreateProfile.tsx

```diff
@@ -61,13 +64,26 @@ export const CreateProfile = ({ onSubmit }: CreateProfileProps) => {
 				<div className="flex flex-row items-center justify-between">
 					<span className="mt-2 text-sm text-theme-neutral-dark">Want to set the wallet to dark mode?</span>
 					<div className="-mt-8">
-						<Toggle />
+						<Toggle ref={register()} name="isDarkMode" />
 					</div>
 				</div>
 			),
 		},
 	];
 
+	const submitForm = async ({ name, currency, isDarkMode, marketProvider }: any) => {
+		const profile = env.profiles().create(name);
+		profile.settings().set(ProfileSetting.MarketProvider, marketProvider);
+		profile.settings().set(ProfileSetting.ChartCurrency, currency);
+		profile.settings().set(ProfileSetting.Theme, isDarkMode ? "dark" : "light");
+
+		await env.persist();
+
+		if (typeof onSubmit === "function") {
```

Comment:
Hi, @alexbarnsley! Just a small doubt :3

Do we really need this `typeof` checking over here? Since we are using TS isn't better to add the right type for the submit (function) and expect to have it otherwise it'll break in compilation time? 🤔

Reply:
Added this more because onSubmit is nullable. I'll change it.